### DISCUSSION TITLE
Proposal for optimizing cache in Umbraco on large-scale .NET Core environments

### DISCUSSION THREAD

**Discussion by landlogicit** [umbraco/Umbraco-CMS#17119] [Features and ideas]
In large-scale Umbraco sites running on .NET Core (from version 9 onwards), managing cache for vast amounts of content can become resource-intensive. To address this, I propose implementing an archive system for older content, which can help significantly reduce memory and CPU usage.

The Idea:
Archive older content: Move content older than 3-4 months into a dedicated table or storage outside of the primary content cache. This would drastically reduce the number of items that need to be cached, freeing up RAM and lowering CPU load, which would also lower operational costs.

SEO and content access: Archiving content doesn’t mean losing SEO benefits. Archived content would still be accessible by intercepting requests through URL routing. The relevant content could be fetched using a custom controller and view, and only then loaded into the cache if needed.

Search and query management: Site search queries should be split between active content and archived content, ensuring optimized performance for both current and older content without burdening the system.

This approach allows us to balance efficient resource usage with the need to maintain a seamless user experience and SEO integrity, especially in scenarios where the volume of content is too large to cache entirely.

Would love to hear your thoughts on this approach!

**Comment by kasparboelkjeldsen** [comment]
I feel like a lot of what you are trying to solve here, will be solved by the new lazy loaded cache in Umbraco 15? 
From https://umbraco.com/blog/umbraco-product-update-august-2024/

"Lazy Loaded Content Cache. This project is internally split into three sub-projects; one for navigation, one for routing, and one for the actual cache handling. When it's done, you'll be able to utilize a hybrid cache and benefit from better performance during boot time and in general benefit from more flexibility related to cache."

**Reply by landlogicit** [reply]
Thank you for the update regarding the lazy-loaded cache in Umbraco 15 – it's great to hear that this feature will offer more flexibility and better performance!

However, for those of us working with earlier versions of Umbraco (9-14), the problem remains, especially for large-scale sites with extensive content. While upgrading to Umbraco 15 might eventually be a solution, upgrading isn't always straightforward, particularly when many third-party components are not yet updated to support the latest version.

For now, the most feasible approach in earlier versions is to archive older content and decouple it from the cache. This strategy helps manage memory and CPU load while maintaining URL accessibility and SEO integrity, without requiring a major system upgrade.

Upgrades can be complex, and sometimes they are not immediately possible in production environments due to dependencies on external packages that haven’t been updated yet. So, having a viable solution for managing cache in previous versions is still essential.

**Comment by nul800sebastiaan** [comment]
Sorry to bring the bad news, but Umbraco 13 is in the support phase and will not be getting new features, nor will any older editions of Umbraco, most of them are EOL or also in the support phase.

You can read about the lifecycle of Umbraco releases here: https://umbraco.com/products/knowledge-center/long-term-support-and-end-of-life/

For Umbraco 13 and below, people have been working around these issues in various ways. If you have this need then I would recommend you ask around on the forum at https://our.umbraco.com or Discord at https://discord.umbraco.com

I'm going to close this discussion as we're focused on providing these types of features in v15 and beyond, as noted in the other discussion: https://github.com/umbraco/Umbraco-CMS/discussions/13734
